src/main/java/com/sfuentes/ClubMember.java

```diff
@@ -0,0 +1,40 @@
+package com.sfuentes;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ClubMember {
+  int memberID;
+  Map<Integer, List<Double>> weightPerScale;
+  double previousWeight;
+
+  public ClubMember(int memberID, double previousWeight, Map<Integer, List<Double>> weightPerScale) {
+
+    this.memberID = memberID;
+    this.previousWeight = previousWeight;
+    this.weightPerScale = weightPerScale;
+  }
+
+  public Map<String, Double> getWeightComparison(double previousWeight, Map<Integer, List<Double>> weightPerScale) {
+    Map<String, Double> comparison = new HashMap<>();
+    double sum = 0.0;
+    double average;
+
+    for (double clave : weightPerScale.get(1)) {
```

Comment:
is the parameter weightPerScale the same as the property? the one you set in the constructor?
if that's the case you shouldn't ask for it in this method, just use the property.

Reply:
fixed